pkg/core/blockchain.go

```diff
@@ -168,6 +169,29 @@ func NewBlockchain(s storage.Store, cfg config.ProtocolConfiguration, log *zap.L
 	return bc, nil
 }
 
+// GetOracleModule returns oracle module if enabled.
+func (bc *Blockchain) GetOracleModule() *oracle.Oracle {
+	if mod, ok := bc.contracts.Oracle.Module.(*oracle.Oracle); ok {
+		return mod
+	}
+	return nil
+}
+
+// InitOracleModule sets oracle module. It doesn't protected by mutex and
+// must be called before `bc.Run()` to avoid data race.
+func (bc *Blockchain) InitOracleModule(cfg config.Config) error {
```

Comment:
This is the only way to reach native oracle contract. There is also a dependency of oracle module on native contract because of oracle response script. Move `OracleRequest` to state and oracle response script to `pkg/oracle`?

Reply:
It we want to provide full config to `NewContracts()`. RPC also needs to get already created instance somehow.

Reply:
Or we may create it as a completely separate module which initializes together with RPC server and blockchain.
Then both blockchain and rpc will have `SetOracle()`.